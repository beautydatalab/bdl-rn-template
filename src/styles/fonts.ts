import globalColors, {themeColor} from 'src/styles/colors';

export const sizes = {
  tiny: 10,
  small: 12,
  medium: 14,
  large: 16,
  extraLarge: 24,
  huge: 40,
};

export const colors = {
  default: globalColors.black,
  disable: globalColors.gray500,
  content: globalColors.gray700,
  theme: themeColor,
};
