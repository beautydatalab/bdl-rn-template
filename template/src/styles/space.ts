export const margins = {
  tiny: 4,
  short: 8,
  medium: 16,
  large: 24,
  huge: 32,
};

export const paddings = {
  tiny: 4,
  short: 8,
  medium: 16,
  large: 24,
  huge: 32,
};
