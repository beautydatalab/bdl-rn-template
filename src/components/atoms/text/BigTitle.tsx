import React from 'react';
import {StyleProp, TextStyle} from 'react-native';
import {colors, sizes} from 'src/styles/fonts';
import styled from 'styled-components/native';

interface Props {
  style?: StyleProp<TextStyle>;
  children?: React.ReactNode;
}

const Title = styled.Text`
  font-size: ${sizes.huge}px;
  color: ${colors.default};
  font-weight: 600;
`;

const BigTitle = ({children, style}: Props) => (
  <Title style={style}>{children}</Title>
);

export default BigTitle;
