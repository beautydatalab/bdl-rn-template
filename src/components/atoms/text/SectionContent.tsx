import React from 'react';
import {StyleProp, TextStyle} from 'react-native';
import {colors, sizes} from 'src/styles/fonts';
import styled from 'styled-components/native';

interface Props {
  style?: StyleProp<TextStyle>;
  children?: React.ReactNode;
}

const Content = styled.Text`
  font-size: ${sizes.large};
  color: ${colors.content};
  font-weight: 400;
`;

const SectionContent = ({children, style}: Props) => (
  <Content style={style}>{children}</Content>
);

export default SectionContent;
