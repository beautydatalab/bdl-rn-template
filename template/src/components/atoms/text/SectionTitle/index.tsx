import React from 'react';
import {StyleProp, TextStyle} from 'react-native';
import {colors, sizes} from 'src/styles/fonts';
import styled from 'styled-components/native';

interface Props {
  style?: StyleProp<TextStyle>;
  children?: React.ReactNode;
}

const Title = styled.Text`
  font-size: ${sizes.extraLarge}px;
  color: ${colors.default};
  font-weight: 600;
`;

export const SectionTitle = ({children, style}: Props) => (
  <Title style={style}>{children}</Title>
);
