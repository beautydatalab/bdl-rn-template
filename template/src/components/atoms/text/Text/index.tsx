import React from 'react';

import {TextProps} from 'react-native';

import styled from 'styled-components/native';

import {colors} from 'src/styles/colors';
import {sizes} from 'src/styles/fonts';

interface Props extends TextProps {
  h1?: boolean;
  h2?: boolean;
  h3?: boolean;
  h4?: boolean;
  color?: string;
  children: React.ReactNode;
}

const TextElement = styled.Text`
  color: ${colors.black};
`;

export const Text = ({
  h1,
  h2,
  h3,
  h4,
  color = colors.black,
  style,
  children,
  ...restProps
}: Props) => {
  const fontSizeStyle = h1
    ? {fontSize: sizes.extraHuge}
    : h2
    ? {fontSize: sizes.huge}
    : h3
    ? {fontSize: sizes.extraLarge}
    : h4
    ? {fontSize: sizes.large}
    : undefined;
  const colorStyle = {color};
  return (
    <TextElement {...restProps} style={[style, fontSizeStyle, colorStyle]}>
      {children}
    </TextElement>
  );
};
