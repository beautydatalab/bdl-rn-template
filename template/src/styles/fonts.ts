import {colors as globalColors, themeColor} from 'src/styles/colors';

export const sizes = {
  tiny: 12,
  small: 14,
  medium: 16,
  large: 20,
  extraLarge: 24,
  huge: 32,
  extraHuge: 40,
};

export const colors = {
  default: globalColors.black,
  disable: globalColors.gray500,
  content: globalColors.gray700,
  theme: themeColor,
};
