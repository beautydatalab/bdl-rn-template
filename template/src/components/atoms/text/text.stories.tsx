import React from 'react';

import {storiesOf} from '@storybook/react-native';
import styled from 'styled-components/native';

import {palette} from 'src/styles/colors';
import {margins} from 'src/styles/space';

import {Text} from './Text';

const CenterView = styled.View`
  flex: 1;
  align-items: center;
  justify-content: center;
`;

const EmptyHeightView = styled.View`
  height: ${margins.medium}px;
`;

storiesOf('text', module)
  .addDecorator((getStory) => <CenterView>{getStory()}</CenterView>)
  .add('Text', () => (
    <>
      <Text h1>Heading 1</Text>
      <Text h2>Heading 2</Text>
      <Text h3>Heading 3</Text>
      <Text h4>Heading 4</Text>
      <EmptyHeightView />
      <Text h3 color={palette.yellow600}>
        <Text h2 color={palette.red500}>
          This is{' '}
        </Text>
        color text
      </Text>
    </>
  ));
