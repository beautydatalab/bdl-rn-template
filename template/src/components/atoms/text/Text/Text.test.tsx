import React from 'react';

import renderer from 'react-test-renderer';

import {Text} from 'src/components/atoms/text/Text';
import {colors} from 'src/styles/colors';

test('render correctly', () => {
  const tree = renderer
    .create(
      <>
        <Text h1>Heading 1</Text>
        <Text h2>Heading 2</Text>
        <Text h3>Heading 3</Text>
        <Text h4>Heading 4</Text>
        <Text h3 color={colors.yellow600}>
          <Text h2 color={colors.red500}>
            This is{' '}
          </Text>
          color text
        </Text>
      </>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
