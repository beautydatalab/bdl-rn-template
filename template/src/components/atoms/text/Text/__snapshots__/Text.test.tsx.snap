// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly 1`] = `
Array [
  <Text
    style={
      Array [
        Object {
          "color": "#212121",
        },
        undefined,
        Object {
          "fontSize": 40,
        },
        Object {
          "color": "#212121",
        },
      ]
    }
  >
    Heading 1
  </Text>,
  <Text
    style={
      Array [
        Object {
          "color": "#212121",
        },
        undefined,
        Object {
          "fontSize": 32,
        },
        Object {
          "color": "#212121",
        },
      ]
    }
  >
    Heading 2
  </Text>,
  <Text
    style={
      Array [
        Object {
          "color": "#212121",
        },
        undefined,
        Object {
          "fontSize": 24,
        },
        Object {
          "color": "#212121",
        },
      ]
    }
  >
    Heading 3
  </Text>,
  <Text
    style={
      Array [
        Object {
          "color": "#212121",
        },
        undefined,
        Object {
          "fontSize": 20,
        },
        Object {
          "color": "#212121",
        },
      ]
    }
  >
    Heading 4
  </Text>,
  <Text
    style={
      Array [
        Object {
          "color": "#212121",
        },
        undefined,
        Object {
          "fontSize": 24,
        },
        Object {
          "color": "#f9ce00",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#212121",
          },
          undefined,
          Object {
            "fontSize": 32,
          },
          Object {
            "color": "#e9494e",
          },
        ]
      }
    >
      This is
       
    </Text>
    color text
  </Text>,
]
`;
